compaction/compaction_manager.cc

```diff
@@ -829,7 +829,7 @@ future<> compaction_manager::stop() {
         return make_ready_future<>();
     } else {
         do_stop();
-        return std::move(*_stop_future);
+        return std::exchange(*_stop_future, make_ready_future<>());
```

Comment:
Why does _stop_future need to be replaced with ready future? The ::stop() method is not re-entrable

Reply:
Perhaps we can do 

```
auto stop_future = std::exchange(_stop_future, std::nullopt);
assert(stop_future);
return std::move(*stop_future);
```

What do you think?

Reply:
Actually, I don't think it's needed as we fcall do_stop() before. Better to leave things as they're. Will only leave that change in do_stop() in addition to the unit test.